src/server.js

```diff
@@ -364,12 +364,10 @@ function initializeClient(socket, client, generateToken, token) {
 					log.error("Failed to update sessions for", client.name, err);
 				}
 			});
-
-			sendInitEvent(token);
 		});
-	} else {
-		sendInitEvent(null);
 	}
+
+	sendInitEvent(token);
```

Comment:
Keep the order, `client.generateToken` has a callback, so it calls `sendInitEvent` inside of it.

Reply:
Good catch, thanks